xclim/indices/_threshold.py

```diff
@@ -1365,3 +1366,74 @@ def tropical_nights(
     return (
         tasmin.pipe(lambda x: (tasmin > thresh) * 1).resample(time=freq).sum(dim="time")
     )
+
+
+@declare_units("", tas="[temperature]", thresh="[temperature]", sum_thresh="K days")
+def degree_days_depassment_date(
+    tas: xarray.DataArray,
+    thresh: str,
+    sum_thresh: str,
+    op: str = ">",
+    after_date: str = None,
+    freq: str = "YS",
+):
+    r"""Degree days depassment date.
+
+    Day of year when the sum of degree days exceeds a threshold. Degree days are
+    computed above or below a given temperature threshold.
+
+    Parameters
+    ----------
+    tas : xarray.DataArray
+      Mean daily temperature.
+    thresh : str
+      Threshold temperature on which to base degree days evaluation.
+    sum_thresh : str
+      Threshold of the degree days sum, in "K days".
+    op : {">", "gt", "<", "lt", ">=", "ge", "<=", "le"}
+      If equivalent to '>', degree days are computed as `tas - thresh` and if
+      equivalent to '<', they are computed as `thresh - tas`.
+    after_date: str, optional
+      If given, the earliest date that can be returned. This acts as a calendar-aware
+      `max(out, after_date)`.
+    freq : str
+      Resampling frequency; Defaults to "YS". If `after_date` is given, `freq` should
+      be annual.
+
+    Returns
+    -------
+    xarray.DataArray
+      Degree days depassment date
+
+    Notes
+    -----
+    Let :math:`TG_{ij}` be the daily mean temperature at day :math:`i` of period :math:`j`,
+    :math:`T` is the reference threshold and :math:`ST` is the sum threshold. Then the
+    degree days depassment date is the first day :math:`k` such that
+
+    .. math::
+
+        \begin{cases}
+        ST < \sum_{i=1}^{k} \max(TG_{ij} - T, 0) & \text{if $op$ is '>'} \\
+        ST < \sum_{i=1}^{k} \max(T - TG_{ij}, 0) & \text{if $op$ is '<'}
+        \end{cases}
+
+    The resulting :math:`k` is expressed as a day of year.
+    """
+    thresh = convert_units_to(thresh, "K")
+    tas = convert_units_to(tas, "K")
+    sum_thresh = convert_units_to(sum_thresh, "K days")
+
+    if op in ["<", "<=", "lt", "le"]:
+        c = thresh - tas
+    elif op in [">", ">=", "gt", "ge"]:
+        c = tas - thresh
+
+    def _depassment_date(grp):
+        return rl.first_run_after_date(
+            grp.cumsum("time") > sum_thresh,
+            window=1,
+            date=after_date,
```

Comment:
I don't think this is doing what is intended with the `after_date`.  My understanding from discussion with interested parties would be that value indicates the beginning of when the `cumsum()` should start accumulating values.  

So I think we would want to assign 'nan' or zeros to `grp`  before `after_date` value then let `rl.first_run_after_date()` run without the `date` input

Reply:
Oh I see! In my approach I thought a user would choose the starting date by passing the appropriate freq, like "AS-JUL" for 1st of july, etc. But I can see how that is limiting.

Reply:
That could work but I heard examples of some applications (Building codes or ??) that would  have super specific dates... "04-15" ... basically not always first of the month